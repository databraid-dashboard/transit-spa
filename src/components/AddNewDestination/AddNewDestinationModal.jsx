import React, { Component } from 'react';
import { Header, Modal } from 'semantic-ui-react';
import AddNewDestinationButton from './AddNewDestinationButton';
import AddNewDestinationForm from './AddNewDestinationForm';

class AddNewDestinationModal extends Component {
  constructor(props) {
    super(props);

    this.state = { modalOpen: false };
  }

  handleOpen = () =>
    this.setState({
      modalOpen: true,
    });

  handleClose = () =>
    this.setState({
      modalOpen: false,
    });

  render() {
    return (
      <Modal
        trigger={
          <AddNewDestinationButton onClick={this.handleOpen}>Show Modal</AddNewDestinationButton>
        }
        open={this.state.modalOpen}
        onClose={this.handleClose}
        basic
        size="small"
      >
        <Header icon="compose" content="Add new destination" />
        <Modal.Content>
          <h3>Enter new destination below</h3>
        </Modal.Content>
        <Modal.Actions>
          <AddNewDestinationForm onClick={this.handleClose} />
        </Modal.Actions>
      </Modal>
    );
  }
}

export default AddNewDestinationModal;
