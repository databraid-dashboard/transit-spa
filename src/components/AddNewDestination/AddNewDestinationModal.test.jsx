import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddNewDestinationModal from './AddNewDestinationModal';

describe('AddNewDestinationModal', () => {
  it('should render modal', () => {
    const component = shallow(<AddNewDestinationModal />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
