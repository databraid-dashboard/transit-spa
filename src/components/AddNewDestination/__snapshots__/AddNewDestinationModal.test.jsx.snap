// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewDestinationModal should render modal 1`] = `
<Modal
  basic={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  onClose={[Function]}
  open={false}
  size="small"
  trigger={
    <AddNewDestinationButton
      onClick={[Function]}
    >
      Show Modal
    </AddNewDestinationButton>
  }
>
  <Header
    content="Add new destination"
    icon="compose"
  />
  <ModalContent>
    <h3>
      Enter new destination below
    </h3>
  </ModalContent>
  <ModalActions>
    <AddNewDestinationForm
      onClick={[Function]}
    />
  </ModalActions>
</Modal>
`;
